src/app/tasks/task.service.ts

```diff
@@ -91,7 +79,7 @@ export class TaskService {
 
     public deleteTask(taskId: number) {
         // Task title is it's unique identifier (needs validation on creation)
-        this.tasks.splice(this.tasks.findIndex(task => task.id == taskId), 1);
+        if (this.tasks) this.tasks.splice(this.tasks.findIndex(task => task.id == taskId), 1);
```

Comment:
this if statement doesn't make sense, because deletion of a task shouldn't be possible if there are no tasks to delete